src/libraries/System.Net.Http/src/System/Net/Http/HttpClientHandler.AnyMobile.cs

```diff
@@ -0,0 +1,429 @@
+        {
+            get => throw new PlatformNotSupportedException();
+            set => throw new PlatformNotSupportedException();
+        }
+
+        [UnsupportedOSPlatform("android")]
+        [UnsupportedOSPlatform("browser")]
+        [UnsupportedOSPlatform("ios")]
+        [UnsupportedOSPlatform("tvos")]
+        [UnsupportedOSPlatform("maccatalyst")]
+        public bool CheckCertificateRevocationList
+        {
+            get => throw new PlatformNotSupportedException();
+            set => throw new PlatformNotSupportedException();
+        }
+
+        [UnsupportedOSPlatform("android")]
+        [UnsupportedOSPlatform("browser")]
+        [UnsupportedOSPlatform("ios")]
+        [UnsupportedOSPlatform("tvos")]
+        [UnsupportedOSPlatform("maccatalyst")]
+        public SslProtocols SslProtocols
+        {
+            get => throw new PlatformNotSupportedException();
+            set => throw new PlatformNotSupportedException();
+        }
+
+        [UnsupportedOSPlatform("android")]
+        [UnsupportedOSPlatform("ios")]
+        [UnsupportedOSPlatform("tvos")]
+        [UnsupportedOSPlatform("maccatalyst")]
+        public IDictionary<string, object?> Properties => throw new PlatformNotSupportedException();
+
+        //
+        // Attributes are commented out due to https://github.com/dotnet/arcade/issues/7585
+        // API compat will fail until this is fixed
+        //
+        //[UnsupportedOSPlatform("android")]
+        [UnsupportedOSPlatform("browser")]
+        //[UnsupportedOSPlatform("ios")]
+        //[UnsupportedOSPlatform("tvos")]
+        //[UnsupportedOSPlatform("maccatalyst")]
+        protected internal override HttpResponseMessage Send(HttpRequestMessage request,
+            CancellationToken cancellationToken)
+        {
+            throw new PlatformNotSupportedException();
+        }
+
+        protected internal override Task<HttpResponseMessage> SendAsync(HttpRequestMessage request,
+            CancellationToken cancellationToken)
+        {
+            if (DiagnosticsHandler.IsGloballyEnabled() && _diagnosticsHandler != null)
+            {
+                return _diagnosticsHandler!.SendAsync(request, cancellationToken);
+            }
+
+            if (IsSocketHandler)
+            {
+                return _socketHandler!.SendAsync(request, cancellationToken);
+            }
+            else
+            {
+                return _underlyingHandler!.SendAsync(request, cancellationToken);
+            }
+        }
+
+        [UnsupportedOSPlatform("android")]
+        [UnsupportedOSPlatform("browser")]
+        [UnsupportedOSPlatform("ios")]
+        [UnsupportedOSPlatform("tvos")]
+        [UnsupportedOSPlatform("maccatalyst")]
+        public static Func<HttpRequestMessage, X509Certificate2?, X509Chain?, SslPolicyErrors, bool> DangerousAcceptAnyServerCertificateValidator =>
+            throw new PlatformNotSupportedException();
```

Comment:
This seems to run counter to the documentation:

> Particularly in test scenarios, a common pattern use HttpClient to connect to a server with a certificate that shouldn't be validated, such as a self-signed certificate. You commonly do this with HttpClientHandler by setting the ServerCertificateCustomValidationCallback property to a delegate that always returns True; this indicates that the certificate has passed validation. However, not all implementations support this callback, and some throw PlatformNotSupportedException.
>
> The DangerousAcceptAnyServerCertificateValidator property addresses this limitation. The delegate returned by the DangerousAcceptAnyServerCertificateValidator property can be assigned to the ServerCertificateCustomValidationCallback property, as the following example does:
>
> ```c#
> handler.ServerCertificateCustomValidationCallback = httpClientHandler.DangerousAcceptAnyServerCertificateValidator;
> ```
>
> This gives HttpClientHandler implementations a known object reference identity that expresses the developer's intention. If the object stored in the DangerousAcceptAnyServerCertificateValidator property is reference equals to DangerousAcceptAnyServerCertificateValidator, the runtime is able to entirely disable validation on a platform that would otherwise throw a PlatformNotSupportedException.

Reply:
We may be able to pull something off w/ the native handlers.  As for `SocketsHttpHandler` (on Android at least), I don't think we'll be able to be able to support it.  Maybe we'll be able to apply an ugly tweak, but as of right now it's not likely.

Reply:
The minimal implementation is to return static `(...) => true` delegate here in `DangerousAcceptAnyServerCertificateValidator` and then check for it by reference in `ServerCertificateCustomValidationCallback` setter and not throw `PlatformNotSupportedException`. At least that's how I read the documentation. It doesn't necessarily guarantee that the validation gets bypassed on platforms where that is not possible. I could be reading the documentation wrong though.

`SockersHttpHandler` on Android may be tricky and I am aware of the limitations in `SslStream` implementation. For something like `NSUrlSessionHandler` it would be possible to expose `DangerousAcceptAnyTrust` callback and assign it through reflection to `TrustOverrideForUrl`.